hipercow-api/Controllers/ClustersController.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Imperial College London. All rights reserved.
+
+namespace Hipercow_api.Controllers
+{
+    using Hipercow_api.Models;
+    using Hipercow_api.Tools;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// The /clusters and /clusters/xxx endpoints provide the list of clusters
+    /// and information about a particular cluster respectively.
+    /// </summary>
+    [ApiController]
+    [Route("api/v1/[controller]")]
+    public class ClustersController : ControllerBase
+    {
+        /// <summary>
+        /// An object implementing IClusterInfoQuery, responsible for returning
+        /// the list of clusters, or information on a specific cluster.
+        /// </summary>
+        private IClusterInfoQuery clusterInfoQuery = new ClusterInfoQuery();
+
+        /// <summary>
+        /// Used for testing. Allow injecting an IClusterInfoQuery object
+        /// returning defined results, rather than doing a genuine query.
+        /// </summary>
+        /// <param name="inject">A mock cluster info query.</param>
+        [NonAction]
+        public void MockClusterInfoQuery(IClusterInfoQuery inject)
```

Comment:
unsure if this is needed if you inject clusterInfoQuery and mock this class in tests

Reply:
Yeah, ideally not. It's almost always possible to find ways to unit test without adding extra test-only functionality to the module under test (though it does require supporting DI etc in the overall design). I think the clusterInfoQuery should be injectable in the constructor?

Reply:
Yep, that disappears with the injection...